compiler/x/codegen/SIMDTreeEvaluator.cpp

```diff
@@ -0,0 +1,237 @@
+      //
+      TR::Register* tempReg = cg->allocateRegister();
+      generateRegMemInstruction(LEARegMem(cg), node, tempReg, mr, cg);
+      mr = generateX86MemoryReference(tempReg, 0, cg);
+      cg->stopUsingRegister(tempReg);
+      }
+   return mr;
+   }
+
+TR::Register* OMR::X86::TreeEvaluator::SIMDRegLoadEvaluator(TR::Node* node, TR::CodeGenerator* cg)
+   {
+   TR::Register* globalReg = node->getRegister();
+   if (!globalReg)
+      {
+      globalReg = cg->allocateRegister(TR_VRF);
+      node->setRegister(globalReg);
+      }
+   return globalReg;
+   }
+
+TR::Register* OMR::X86::TreeEvaluator::SIMDRegStoreEvaluator(TR::Node* node, TR::CodeGenerator* cg)
+   {
+   TR::Node* child = node->getFirstChild();
+   TR::Register* globalReg = cg->evaluate(child);
+   cg->machine()->setXMMGlobalRegister(node->getGlobalRegisterNumber() - cg->machine()->getNumGlobalGPRs(), globalReg);
+   cg->decReferenceCount(child);
+   return globalReg;
+   }
+
+TR::Register* OMR::X86::TreeEvaluator::SIMDloadEvaluator(TR::Node* node, TR::CodeGenerator* cg)
+   {
+   TR::MemoryReference* tempMR = generateX86MemoryReference(node, cg);
+   tempMR = ConvertToPatchableMemoryReference(tempMR, node, cg);
+   TR::Register* resultReg = cg->allocateRegister(TR_VRF);
+
+   TR_X86OpCodes opCode = BADIA32Op;
+   switch (node->getSize())
+      {
+      case 16:
+         opCode = MOVDQURegMem;
+         break;
+      default:
+         break;
+      }
+   TR_ASSERT(opCode != BADIA32Op, "SIMD Load Evaluator: unsupported fill size");
+
+   TR::Instruction* instr = generateRegMemInstruction(opCode, node, resultReg, tempMR, cg);
+   if (node->getOpCode().isIndirect())
+      cg->setImplicitExceptionPoint(instr);
+   node->setRegister(resultReg);
+   tempMR->decNodeReferenceCounts(cg);
+   return resultReg;
+   }
+
+TR::Register* OMR::X86::TreeEvaluator::SIMDstoreEvaluator(TR::Node* node, TR::CodeGenerator* cg)
+   {
+   TR::Node* valueNode = node->getChild(node->getOpCode().isIndirect() ? 1 : 0);
+   TR::MemoryReference* tempMR = generateX86MemoryReference(node, cg);
+   tempMR = ConvertToPatchableMemoryReference(tempMR, node, cg);
+   TR::Register* valueReg = cg->evaluate(valueNode);
+
+   TR_X86OpCodes opCode = BADIA32Op;
+   switch (node->getSize())
+      {
+      case 16:
+         opCode = MOVDQUMemReg;
+         break;
+      default:
+         break;
+      }
+   TR_ASSERT(opCode != BADIA32Op, "SIMD Store Evaluator: unsupported fill size");
+
+   TR::Instruction* instr = generateMemRegInstruction(opCode, node, tempMR, valueReg, cg);
+
+   cg->decReferenceCount(valueNode);
+   tempMR->decNodeReferenceCounts(cg);
+   if (node->getOpCode().isIndirect())
+      cg->setImplicitExceptionPoint(instr);
+   return NULL;
+   }
+
+TR::Register* OMR::X86::TreeEvaluator::SIMDsplatsEvaluator(TR::Node* node, TR::CodeGenerator* cg)
+   {
+   TR::Node* childNode = node->getChild(0);
+   TR::Register* childReg = cg->evaluate(childNode);
+
+   uint8_t shufconst = 0;
```

Comment:
This deserves a comment and what is the hardcoded 0x44 - it seems a bit of a magic number - should it be in a constant or something with a name to make it more sensible.